runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/impl/StandardRetryPolicy.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.retries.impl
+
+import aws.smithy.kotlin.runtime.ClientException
+import aws.smithy.kotlin.runtime.SdkBaseException
+import aws.smithy.kotlin.runtime.ServiceException
+import aws.smithy.kotlin.runtime.ServiceException.*
+import aws.smithy.kotlin.runtime.retries.RetryDirective
+import aws.smithy.kotlin.runtime.retries.RetryErrorType
+import aws.smithy.kotlin.runtime.retries.RetryPolicy
+
+/**
+ * A standard retry policy which attempts to derive information from the Smithy exception hierarchy.
+ */
+open class StandardRetryPolicy : RetryPolicy<Any?> {
```

Comment:
question/comment 
I'm not sure this should be in `impl`. It seems like something that is part of the public interface that someone may want to re-use or specify themselves?

Reply:
Yes, `StandardRetryPolicy` is part of the public interface (in the non-`interface` sense of the word) but as I alluded elsewhere, the `impl` namespace is not intended to be internal-only...it's intended to hold implementations.

Is the distinction between interface and implementation at the namespace level perhaps not useful here?

Reply:
Now that I understand what you were shooting for it makes a _little_ more sense. 

> Is the distinction between interface and implementation at the namespace level perhaps not useful here?

Thinking about it for a second I don't know that we gain much from it. I don't know of anywhere else we do this either. I think _if_ we wanted more taxonomy it would make more sense to have subpackages according to type. e.g.

```
retries/
    strategy/
    policy/
```